src/main/perf/IndexThreads.java

```diff
@@ -273,12 +287,12 @@ public boolean hasNext() {
                         Field extraField;
 
                         try {
-                          doc = docs.nextDoc(docState);
+                          doc = docs.nextDoc(docState, true);
```

Comment:
Hmm why do we know this should always be `expected=true`?  Couldn't the very last group have an ob1 and expect more docs than are really left?

Reply:
Good question, I was assuming the doc allocated per group was accurate, and after thinking a bit, I think it is accurate:

The process of deciding `numDocs` per group is
```
if (groupCounter == groupBlocks.length-1) {
  numDocs = numTotalDocs - ((int) (groupCounter*docsPerGroupBlock));
} else {
  numDocs = ((int) ((1+groupCounter)*docsPerGroupBlock)) - ((int) (groupCounter*docsPerGroupBlock));
}
```
And then if we sum all the `numDocs` over the group index we will get: (assuming we have `G` groups)
```
numTotalDocs - ((int) ((G-1)*docsPerGroupBlock)) + 
((int) ((G-1)*docsPerGroupBlock)) - ((int) ((G-2)*docsPerGroupBlock)) +
...
((int) (docsPerGroupBlock)) - 0
 = 
numTotalDocs
```
So I think we'll have exact `numTotalDocs` indexed at last.

Reply:
Also I just tested using my 10k binary LFD with 423 groups, and it seems it is safe

Reply:
LOL OK!